lib/money/currency.rb

```diff
@@ -0,0 +1,72 @@
+class Money
+  class Currency
+    class UnknownCurrency < ArgumentError; end
+
+    CURRENCY_DATA_PATH = "#{File.expand_path('../../../config', __FILE__)}/currency_iso.json".freeze
+
+    @@mutex = Mutex.new
+
+    attr_reader :priority, :iso_code, :iso_numeric, :name, :symbol,
+      :disambiguate_symbol, :html_entity, :subunit, :subunit_to_unit, :decimal_mark,
+      :thousands_separator, :symbol_first, :smallest_denomination, :alternate_symbols
+
+    class << self
+      def new(currency_iso)
+        iso = currency_iso.to_s.downcase
+        loaded_currencies[iso] || @@mutex.synchronize { loaded_currencies[iso] = super(iso) }
+      end
+      alias_method :find, :new
+
+      def loaded_currencies
+        @loaded_currencies ||= {}
+      end
+
+      def currencies_json
+        @@currencies_json ||= begin
+          json = File.read(CURRENCY_DATA_PATH)
+          json.force_encoding(::Encoding::UTF_8) if defined?(::Encoding)
+          JSON.parse(json)
+        end
+      end
+    end
+    currencies_json
+
+    def initialize(currency_iso)
+      return nil if currency_iso.blank?
+      unless data = self.class.currencies_json[currency_iso]
+        raise UnknownCurrency, "Unknown currency '#{currency_iso}'"
+      end
+      @alternate_symbols     = data['alternate_symbols']
+      @decimal_mark          = data['decimal_mark']
```

Comment:
As discussed today we should remove this and thousands_separator as these are properties from the locale and not from the currency.